models/ed/R/veg2model.ED2.R

```diff
@@ -33,32 +52,26 @@ veg2model.ED2 <- function(outfolder, start_date, end_date,
   #--------------------------------------------------------------------------------------------------#
   # Prepare pss
   
-  # get data that was processed in the upstream 
-  # this check should be unnecessary once we handle pss data in the upstream
-  if(source == "FIA"){
-    
-    # this might change depending on how I process data in put.veg  
-    pss <- veg_info[[1]]
-    
-  } else{
-    
-    # other cases for now,
-    # create pss file from scratch by using values passed from using defaults 
-    # if you don't want to pass these from settings make sure you pass them to convert.inputs in put.veg.module
-    # wait until we solve metadata problem?
-    
-    ### assuming one patch for now, otherwise these lines might change ###
-    time    <- start_year-1
-    n.patch <- 1
-    trk     <- 1
-    age     <- 100
+  # veg_info[[1]] either has 
+  # i)   full pss info (FIA case)
+  # ii)  info passed from settings
+  # iii) no info
+  pss <- as.data.frame(veg_info[[1]], stringsAsFactors = FALSE)
+  
+  # for FIA these steps are unnecessary, it already has the pss info
+  if(source != "FIA"){
+    time    <- ifelse(!is.null(pss$time), pss$time, start_year)
+    n.patch <- ifelse(!is.null(pss$n.patch), pss$n.patch, 1)
```

Comment:
Yep, we already have to designate a column in veg_info[[2]] to indicate plot/patch designations and we can use that to count up the number of unique plots.